cetmix_tower_server/tests/test_key.py

```diff
@@ -72,10 +63,97 @@ def test_key_access_rights(self):
         with self.assertRaises(AccessError):
             key_bob.unlink()
 
-        # Add Bob to Root group and test write again
+        # Add Bob to Root group and test delete again
         self.add_to_group(self.user_bob, "cetmix_tower_server.group_root")
         key_bob.unlink()
 
+    def test_server_key_access_rights(self):
+        """Test server private key security features"""
+
+        # Store server key value
+        self.write_and_invalidate(
+            self.key_1, **{"secret_value": "pepe", "server_id": self.server_test_1.id}
+        )
+
+        # Get key value as Bob
+        key_bob = self.key_1.with_user(self.user_bob)
+
+        # Add user to group
+        self.add_to_group(self.user_bob, "cetmix_tower_server.group_user")
+
+        with self.assertRaises(AccessError):
+            # Get value
+            key_bob.read(["secret_value"])
+
+        # Test write
+        with self.assertRaises(AccessError):
+            self.write_and_invalidate(key_bob, **{"secret_value": "frog"})
+
+        # Add Bob as subscriber to server
+        self.server_test_1.message_subscribe([self.user_bob.partner_id.id])
+
+        # User cannot read key value
+        with self.assertRaises(AccessError):
+            key_bob.read(["secret_value"])
+
+        # Remove Bob from server followers
+        self.server_test_1.message_unsubscribe([self.user_bob.partner_id.id])
+
+        # Add Bob to Manager group
+        self.add_to_group(self.user_bob, "cetmix_tower_server.group_manager")
+
+        # Manager cannot read key value
+        with self.assertRaises(AccessError):
+            key_bob.read(["secret_value"])
+
+        # Add Bob as subscriber to server
+        self.server_test_1.message_subscribe([self.user_bob.partner_id.id])
+
+        # Now manager server subscriber can read and update this key
+        key_bob.read(["secret_value"])
```

Comment:
But which value does he get? Plain text password, or a placeholder?
Should't we try to write first before reading, so we could ensure that the secret values is properly masked?